android/src/onramp/java/com/reactnativestripesdk/OnrampSdkModule.kt

```diff
@@ -622,6 +623,26 @@ class OnrampSdkModule(
     }
   }
 
+  @ReactMethod
+  override fun authenticateUserWithToken(
+    token: String,
+    promise: Promise,
+  ) {
+    val coordinator =
+      onrampCoordinator ?: run {
+        promise.resolve(createOnrampNotConfiguredError())
+        return
+      }
+
+    CoroutineScope(Dispatchers.IO).launch {
+      val result = coordinator.authenticateUserWithToken(token)
+
+      CoroutineScope(Dispatchers.Main).launch {
+        handleAuthenticateUserWithTokenResult(result, promise)
+      }
```

Comment:
I don't think you need to launch here
```suggestion
      withContext(Dispatchers.Main) {
        handleAuthenticateUserWithTokenResult(result, promise)
      }
```